examples/all-clusters-app/all-clusters-common/src/operational-state-delegates.cpp

```diff
@@ -0,0 +1,184 @@
+ * Note: User Define
+ */
+constexpr DelegatesEnquiryTable kDelegatesEnquiryTable[] = {
+    // EndpointId, ClusterId, Delegate
+    { Clusters::OperationalState::kDemoEndpointId, Clusters::OperationalState::Id, &opStateDelegate },
+};
+
+/**
+ * Get the pointer of target delegate for target endpoint and cluster
+ * @param[in] aEndpiontId The endpointId
+ * @param[in] aClusterID  The clusterId
+ * @return the pointer of target delegate
+ */
+Delegate * getGenericDelegateTable(EndpointId aEndpointId, ClusterId aClusterId)
+{
+    for (size_t i = 0; i < ArraySize(kDelegatesEnquiryTable); ++i)
+    {
+        if (kDelegatesEnquiryTable[i].mEndpointId == aEndpointId && kDelegatesEnquiryTable[i].mClusterId == aClusterId)
+        {
+            return kDelegatesEnquiryTable[i].pItems;
+        }
+    }
+    return nullptr;
+}
+
+// @brief Instance getter for the default global delegate for operational state alias cluster
+// The delegate API assumes there will be separate delegate objects for each cluster instance.
+// (i.e. each separate operational state cluster derivation, on each separate endpoint)
+// @note This API should always be called prior to using the delegate and the return pointer should never be cached.
+// @return Default global delegate instance.
+Delegate * GetOperationalStateDelegate(EndpointId endpointId, ClusterId clusterId)
+{
+    return getGenericDelegateTable(endpointId, clusterId);
+}
+
+/**
+ * Log OperationalError Event
+ * @param[in] aEndpointId Target endpointId to log event
+ * @param[in] err Set the operational error to event.
+ * @return true: send event success; fail : send event fail.
+ */
+bool LogOperationalErrorEvent(EndpointId aEndpointId, const GenericOperationalError & err)
+{
+    Events::OperationalError::Type event;
+    EventNumber eventNumber;
+
+    event.errorState = err;
+    CHIP_ERROR error = app::LogEvent(event, aEndpointId, eventNumber);
+
+    if (error != CHIP_NO_ERROR)
+    {
+        return false;
+    }
+
+    // notify operational state need to change
+    MatterReportingAttributeChangeCallback(aEndpointId, event.GetClusterId(), OperationalState::Attributes::OperationalState::Id);
+    return true;
+}
+
+/**
+ * Log OperationalCompletion Event
+ * @param[in] aEndpointId Target endpointId to log event
+ * @param[in] op Set the operational completion to event.
+ * @return true: send event success; fail : send event fail.
+ */
+bool LogOperationCompletion(EndpointId aEndpointId, const GenericOperationCompletion & op)
+{
+    Events::OperationCompletion::Type event;
+    EventNumber eventNumber;
+    event = op;
+
+    CHIP_ERROR error = app::LogEvent(event, aEndpointId, eventNumber);
+
+    if (error != CHIP_NO_ERROR)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+} // namespace OperationalState
+} // namespace Clusters
+} // namespace app
+} // namespace chip
+
+void MatterOperationalStatePluginServerInitCallback()
```

Comment:
This is expected to be implemented by the cluster implementation itself, not the application.  We need a different mechanism for applications initializing OperationalStateServer instances... Probably during application startup or when setting up the relevant dynamic endpoints.